spec/app/github_repo_fetcher_spec.rb

```diff
@@ -122,80 +122,81 @@ def docs_url(repo_name)
       "https://api.github.com/repos/alphagov/#{repo_name}/contents/docs"
     end
 
-    def with_stubbed_client(temporary_client, instance)
-      before_client = instance.instance_variable_get(:@client)
-      instance.instance_variable_set(:@client, temporary_client)
-      yield
-      instance.instance_variable_set(:@client, before_client)
-    end
-
-    it "returns an array of hashes including title derived from markdown contents" do
-      commit = { sha: SecureRandom.hex(40), timestamp: Time.now.utc.to_s }
-      doc_contents = "# title \n Some document"
-      doc = double("doc", type: "file", name: "foo.md", download_url: "foo_url", path: "docs/foo.md", html_url: "foo_html_url")
-
-      instance = GitHubRepoFetcher.new
-      allow(instance).to receive(:latest_commit).and_return(commit)
-      allow(instance).to receive(:repo).with(repo_name).and_return(public_repo)
-      allow(HTTP).to receive(:get).with(doc.download_url).and_return(doc_contents)
-      with_stubbed_client(double("Octokit::Client", contents: [doc]), instance) do
-        expect(instance.docs(repo_name)).to eq([
-          {
-            title: "title",
-            path: "/apps/#{repo_name}/foo.html",
-            markdown: doc_contents,
-            relative_path: "docs/foo.md",
-            source_url: "foo_html_url",
-            latest_commit: commit,
-          },
-        ])
+    context "the repo contains a reachable docs/ folder" do
+      def with_stubbed_client(temporary_client, instance)
+        before_client = instance.instance_variable_get(:@client)
+        instance.instance_variable_set(:@client, temporary_client)
+        yield
+        instance.instance_variable_set(:@client, before_client)
       end
-    end
-
-    it "retrieves documents recursively" do
-      dir = double("dir", type: "dir", name: "foo", path: "docs/foo")
-      nested_doc = double("nested_doc", type: "file", name: "bar.md", path: "docs/foo/bar.md", download_url: "bar_url", html_url: "bar_html_url")
 
-      doc_contents = "# title \n Some document"
-      commit = { sha: SecureRandom.hex(40), timestamp: Time.now.utc.to_s }
-      instance = GitHubRepoFetcher.new
-      allow(instance).to receive(:latest_commit).and_return(commit)
-      allow(instance).to receive(:repo).with(repo_name).and_return(public_repo)
-      allow(HTTP).to receive(:get).with(nested_doc.download_url).and_return(doc_contents)
+      let(:commit) { { sha: SecureRandom.hex(40), timestamp: Time.now.utc.to_s } }
+      let(:instance) do
+        instance = GitHubRepoFetcher.new
+        allow(instance).to receive(:latest_commit).and_return(commit)
+        allow(instance).to receive(:repo).with(repo_name).and_return(public_repo)
+        instance
+      end
+      let(:doc_contents) { "# title \n Some document" }
+      let(:doc_title_derived_from_contents) { "title" }
+
+      it "returns an array of hashes including title derived from markdown contents" do
```

Comment:
Yeah, this has always been a bit of a smell. I think it's because we're testing two things: 1) that it returns an array of hashes, and 2) that the title is derived from the markdown contents (which isn't tested anywhere else).

I'll have a think about splitting it out into two tests, which would be more appropriate.

Reply:
That looks neater 👍. I noticed we're no longer testing some of the content of the hash, though, which seems like a regression, especially since we still check it for the nested case. Should we check it still?